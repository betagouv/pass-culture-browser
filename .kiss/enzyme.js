import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
