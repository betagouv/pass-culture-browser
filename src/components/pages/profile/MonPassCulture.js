/* eslint
  react/jsx-one-expression-per-line: 0 */
import React from 'react'
import PropTypes from 'prop-types'

import { getAvailableBalanceByType } from './utils'

const MonPassCulture = ({ user }) => {
  const { expenses } = user
  const allWallet =
    (typeof user.wallet_balance !== 'number' && '--') || user.wallet_balance
  const { digital, physical } = expenses
  const [digitalAvailable, physicalAvailable] = [digital, physical].map(
    getAvailableBalanceByType(allWallet)
  )
  return (
    <div id="profile-page-user-wallet" className="jauges padded">
      <div className="text overall flex-1">
        <b className="is-block" id="profile-wallet-balance-value">
          {`Il reste ${allWallet} €`}
        </b>
        <span className="is-block fs14">sur votre Pass Culture</span>
      </div>
      <div className="text-containers flex-0 mt12 py12 mr8">
        <div className="fs14">
          <span className="is-block" id="profile-physical-wallet-value">
            Jusqu&apos;à <b>{physicalAvailable} €</b> pour les biens culturels
          </span>
        </div>
        <div className="fs14 mt12">
          <span className="is-block" id="profile-digital-wallet-value">
            Jusqu&apos;à <b>{digitalAvailable} €</b> pour les offres numériques
          </span>
        </div>
      </div>
    </div>
  )
}

MonPassCulture.propTypes = {
  user: PropTypes.object.isRequired,
}

export default MonPassCulture
