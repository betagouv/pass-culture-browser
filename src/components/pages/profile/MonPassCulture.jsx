import PropTypes from 'prop-types'
import React from 'react'

import { getAvailableBalanceByType } from './utils'
import { getWalletValue } from '../../../utils/user'
import { formatEndValidityDate } from '../../../utils/date/date'

const MonPassCulture = ({ currentUser }) => {
  const { expenses, wallet_date_created } = currentUser || {}
  const allWallet = getWalletValue(currentUser)
  const { digital, physical } = expenses
  const [digitalAvailable, physicalAvailable] = [digital, physical].map(
    getAvailableBalanceByType(allWallet)
  )
  let endValidityDate = null
  if (wallet_date_created) {
    endValidityDate = formatEndValidityDate(new Date(wallet_date_created))
  }

  return (
    <div
      className="jauges padded  text-center"
      id="profile-page-user-wallet"
    >
      <div className="text overall">
        <b
          className="is-block"
          id="profile-wallet-balance-value"
        >
          {`Il reste ${allWallet} €`}
        </b>
        <span className="is-block fs14">{'sur votre pass Culture'}</span>
      </div>
      <div className="text-containers mt12 py12 mr8">
        <div className="fs15">
          <span
            className="is-block"
            id="profile-physical-wallet-value"
          >
            {'Jusqu’à '}
            <b>{`${physicalAvailable} €`}</b>
            {' pour les biens culturels'}
          </span>
        </div>
        <div className="fs15 mt12">
          <span
            className="is-block"
            id="profile-digital-wallet-value"
          >
            {'Jusqu’à '}
            <b>{`${digitalAvailable} €`}</b>
            {' pour les offres numériques'}
          </span>
        </div>
        {endValidityDate && (
          <div className="fs15 mt12">
            <span
              className="is-block"
              id="profile-end-validity-date"
            >
              {`Votre crédit est valable jusqu’au ${endValidityDate}.`}
            </span>
          </div>
        )}
      </div>
    </div>
  )
}

MonPassCulture.propTypes = {
  currentUser: PropTypes.shape().isRequired,
}

export default MonPassCulture
