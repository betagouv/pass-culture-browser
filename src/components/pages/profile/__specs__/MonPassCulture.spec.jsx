// jest --env=jsdom ./src/components/pages/profile/tests/MonPassCulture --watch
import React from 'react'
import { shallow } from 'enzyme'

import MonPassCulture from '../MonPassCulture'

const walletId = '#profile-wallet-balance-value'
const digitalId = '#profile-digital-wallet-value'
const physicalId = '#profile-physical-wallet-value'

describe('src | components | MonPassCulture', () => {
  describe('snapshot', () => {
    it('should match snapshot with required props', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 0, max: 200 },
            physical: { actual: 0, max: 200 },
          },
          wallet_balance: 500,
        },
      }

      // when
      const wrapper = shallow(<MonPassCulture {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
  describe('render', () => {
    it('with wallet value set to 0', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 100, max: 200 },
            physical: { actual: 100, max: 200 },
          },
          wallet_balance: 0,
        },
      }

      // when
      const wrapper = shallow(<MonPassCulture {...props} />)
      const walletElement = wrapper.find(walletId)
      const digitalElement = wrapper.find(digitalId)
      const physicalElement = wrapper.find(physicalId)

      // then
      let expected = `Il reste 0 €`
      let text = walletElement.text()
      expect(text).toEqual(expected)

      expected = `Jusqu'à 0 € pour les offres numériques`
      text = digitalElement.text()
      expect(digitalElement.text()).toEqual(expected)
      expected = `Jusqu'à 0 € pour les biens culturels`

      text = physicalElement.text()
      expect(physicalElement.text()).toEqual(expected)
    })
    it('with wallet value set to 90', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 20, max: 200 },
            physical: { actual: 80, max: 200 },
          },
          wallet_balance: 90,
        },
      }

      // when
      const wrapper = shallow(<MonPassCulture {...props} />)
      const walletElement = wrapper.find(walletId)
      const digitalElement = wrapper.find(digitalId)
      const physicalElement = wrapper.find(physicalId)

      // then
      let expected = `Il reste 90 €`
      let text = walletElement.text()
      expect(text).toEqual(expected)

      expected = `Jusqu'à 90 € pour les offres numériques`
      text = digitalElement.text()
      expect(digitalElement.text()).toEqual(expected)
      expected = `Jusqu'à 90 € pour les biens culturels`

      text = physicalElement.text()
      expect(physicalElement.text()).toEqual(expected)
    })
    it('render with wallet value set to 200', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 120, max: 200 },
            physical: { actual: 140, max: 200 },
          },
          wallet_balance: 200,
        },
      }

      // when
      const wrapper = shallow(<MonPassCulture {...props} />)
      const walletElement = wrapper.find(walletId)
      const digitalElement = wrapper.find(digitalId)
      const physicalElement = wrapper.find(physicalId)

      // then
      let expected = `Il reste 200 €`
      let text = walletElement.text()
      expect(text).toEqual(expected)

      expected = `Jusqu'à 80 € pour les offres numériques`
      text = digitalElement.text()
      expect(digitalElement.text()).toEqual(expected)

      expected = `Jusqu'à 60 € pour les biens culturels`
      text = physicalElement.text()
      expect(physicalElement.text()).toEqual(expected)
    })
    it('render with invalid wallet balance', () => {
      // given
      const props = {
        currentUser: {
          expenses: {
            digital: { actual: 120, max: 200 },
            physical: { actual: 140, max: 200 },
          },
          wallet_balance: null,
        },
      }

      // when
      const wrapper = shallow(<MonPassCulture {...props} />)
      const walletElement = wrapper.find(walletId)
      const digitalElement = wrapper.find(digitalId)
      const physicalElement = wrapper.find(physicalId)

      // then
      let expected = `Il reste -- €`
      let text = walletElement.text()
      expect(text).toEqual(expected)

      expected = `Jusqu'à -- € pour les offres numériques`
      text = digitalElement.text()
      expect(text).toEqual(expected)

      expected = `Jusqu'à -- € pour les biens culturels`
      text = physicalElement.text()
      expect(text).toEqual(expected)
    })
  })
})
