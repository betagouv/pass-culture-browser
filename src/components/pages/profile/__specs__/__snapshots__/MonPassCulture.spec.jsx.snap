// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | MonPassCulture should match snapshot with required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MonPassCulture
    currentUser={
      Object {
        "expenses": Object {
          "digital": Object {
            "actual": 0,
            "max": 200,
          },
          "physical": Object {
            "actual": 0,
            "max": 200,
          },
        },
        "wallet_balance": 500,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="text overall flex-1"
        >
          <b
            className="is-block"
            id="profile-wallet-balance-value"
          >
            Il reste 500 €
          </b>
          <span
            className="is-block fs14"
          >
            sur votre pass Culture
          </span>
        </div>,
        <div
          className="text-containers flex-0 mt12 py12 mr8"
        >
          <div
            className="fs14"
          >
            <span
              className="is-block"
              id="profile-physical-wallet-value"
            >
              Jusqu’à
               
              <b>
                200
                 
                €
              </b>
               
              pour les biens culturels
            </span>
          </div>
          <div
            className="fs14 mt12"
          >
            <span
              className="is-block"
              id="profile-digital-wallet-value"
            >
              Jusqu’à
               
              <b>
                200
                 
                €
              </b>
               
              pour les offres numériques
            </span>
          </div>
        </div>,
      ],
      "className": "jauges padded",
      "id": "profile-page-user-wallet",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <b
              className="is-block"
              id="profile-wallet-balance-value"
            >
              Il reste 500 €
            </b>,
            <span
              className="is-block fs14"
            >
              sur votre pass Culture
            </span>,
          ],
          "className": "text overall flex-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Il reste 500 €",
              "className": "is-block",
              "id": "profile-wallet-balance-value",
            },
            "ref": null,
            "rendered": "Il reste 500 €",
            "type": "b",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "sur votre pass Culture",
              "className": "is-block fs14",
            },
            "ref": null,
            "rendered": "sur votre pass Culture",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fs14"
            >
              <span
                className="is-block"
                id="profile-physical-wallet-value"
              >
                Jusqu’à
                 
                <b>
                  200
                   
                  €
                </b>
                 
                pour les biens culturels
              </span>
            </div>,
            <div
              className="fs14 mt12"
            >
              <span
                className="is-block"
                id="profile-digital-wallet-value"
              >
                Jusqu’à
                 
                <b>
                  200
                   
                  €
                </b>
                 
                pour les offres numériques
              </span>
            </div>,
          ],
          "className": "text-containers flex-0 mt12 py12 mr8",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="is-block"
                id="profile-physical-wallet-value"
              >
                Jusqu’à
                 
                <b>
                  200
                   
                  €
                </b>
                 
                pour les biens culturels
              </span>,
              "className": "fs14",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Jusqu’à",
                  " ",
                  <b>
                    200
                     
                    €
                  </b>,
                  " ",
                  "pour les biens culturels",
                ],
                "className": "is-block",
                "id": "profile-physical-wallet-value",
              },
              "ref": null,
              "rendered": Array [
                "Jusqu’à",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      200,
                      " ",
                      "€",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    200,
                    " ",
                    "€",
                  ],
                  "type": "b",
                },
                " ",
                "pour les biens culturels",
              ],
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="is-block"
                id="profile-digital-wallet-value"
              >
                Jusqu’à
                 
                <b>
                  200
                   
                  €
                </b>
                 
                pour les offres numériques
              </span>,
              "className": "fs14 mt12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Jusqu’à",
                  " ",
                  <b>
                    200
                     
                    €
                  </b>,
                  " ",
                  "pour les offres numériques",
                ],
                "className": "is-block",
                "id": "profile-digital-wallet-value",
              },
              "ref": null,
              "rendered": Array [
                "Jusqu’à",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      200,
                      " ",
                      "€",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    200,
                    " ",
                    "€",
                  ],
                  "type": "b",
                },
                " ",
                "pour les offres numériques",
              ],
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="text overall flex-1"
          >
            <b
              className="is-block"
              id="profile-wallet-balance-value"
            >
              Il reste 500 €
            </b>
            <span
              className="is-block fs14"
            >
              sur votre pass Culture
            </span>
          </div>,
          <div
            className="text-containers flex-0 mt12 py12 mr8"
          >
            <div
              className="fs14"
            >
              <span
                className="is-block"
                id="profile-physical-wallet-value"
              >
                Jusqu’à
                 
                <b>
                  200
                   
                  €
                </b>
                 
                pour les biens culturels
              </span>
            </div>
            <div
              className="fs14 mt12"
            >
              <span
                className="is-block"
                id="profile-digital-wallet-value"
              >
                Jusqu’à
                 
                <b>
                  200
                   
                  €
                </b>
                 
                pour les offres numériques
              </span>
            </div>
          </div>,
        ],
        "className": "jauges padded",
        "id": "profile-page-user-wallet",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b
                className="is-block"
                id="profile-wallet-balance-value"
              >
                Il reste 500 €
              </b>,
              <span
                className="is-block fs14"
              >
                sur votre pass Culture
              </span>,
            ],
            "className": "text overall flex-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Il reste 500 €",
                "className": "is-block",
                "id": "profile-wallet-balance-value",
              },
              "ref": null,
              "rendered": "Il reste 500 €",
              "type": "b",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sur votre pass Culture",
                "className": "is-block fs14",
              },
              "ref": null,
              "rendered": "sur votre pass Culture",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="fs14"
              >
                <span
                  className="is-block"
                  id="profile-physical-wallet-value"
                >
                  Jusqu’à
                   
                  <b>
                    200
                     
                    €
                  </b>
                   
                  pour les biens culturels
                </span>
              </div>,
              <div
                className="fs14 mt12"
              >
                <span
                  className="is-block"
                  id="profile-digital-wallet-value"
                >
                  Jusqu’à
                   
                  <b>
                    200
                     
                    €
                  </b>
                   
                  pour les offres numériques
                </span>
              </div>,
            ],
            "className": "text-containers flex-0 mt12 py12 mr8",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="is-block"
                  id="profile-physical-wallet-value"
                >
                  Jusqu’à
                   
                  <b>
                    200
                     
                    €
                  </b>
                   
                  pour les biens culturels
                </span>,
                "className": "fs14",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Jusqu’à",
                    " ",
                    <b>
                      200
                       
                      €
                    </b>,
                    " ",
                    "pour les biens culturels",
                  ],
                  "className": "is-block",
                  "id": "profile-physical-wallet-value",
                },
                "ref": null,
                "rendered": Array [
                  "Jusqu’à",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        200,
                        " ",
                        "€",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      200,
                      " ",
                      "€",
                    ],
                    "type": "b",
                  },
                  " ",
                  "pour les biens culturels",
                ],
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="is-block"
                  id="profile-digital-wallet-value"
                >
                  Jusqu’à
                   
                  <b>
                    200
                     
                    €
                  </b>
                   
                  pour les offres numériques
                </span>,
                "className": "fs14 mt12",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Jusqu’à",
                    " ",
                    <b>
                      200
                       
                      €
                    </b>,
                    " ",
                    "pour les offres numériques",
                  ],
                  "className": "is-block",
                  "id": "profile-digital-wallet-value",
                },
                "ref": null,
                "rendered": Array [
                  "Jusqu’à",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        200,
                        " ",
                        "€",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      200,
                      " ",
                      "€",
                    ],
                    "type": "b",
                  },
                  " ",
                  "pour les offres numériques",
                ],
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
